docs/security.md

```diff
@@ -1,5 +1,54 @@
 # Security
 
+## Workflow Controller Security
+
+This has three parts.
+
+### Controller Permissions
+
+The controller has permission (via Kubernetes RBAC + its config map) with either all namespaces (cluster-scope install) or a single [managed namespace](managed-namespace.md) (namespace-install), notably:
+
+* List/get/update workflows, and cron-workflows.
+* Create/get/delete pods, PVCs, and PDBs.
+* List/get template, config maps, service accounts, and secrets.
+
+See [workflow-controller-clusterrole.yaml](manifests/cluster-install/workflow-controller-rbac/workflow-controller-clusterrole.yaml) or [workflow-controller-role.yaml](manifests/namespace-install/workflow-controller-rbac/workflow-controller-role.yaml)
+
+### User Permissions
+
+Users minimally need permission to create/read workflows. The controller will then create workflow pods (config maps etc) on behalf of the users, even if the user does not have permission to do this themselves. The controller will only create workflow pods in the workflow's namespace.
+
+A way to think of this is that, if the user has permission to create a workflow in a namespace, then it is OK to create pods or anything else for them in that namespace.
+
+If the user only has permission to create workflows, then they will be typically unable to configure other necessary resources such as config maps, or view the outcome of their workflow. This is useful when the user is a service.  
+
+!!! Warning
+    If you allow users to create workflows in the controller's namespace (typically `argo`), it may be possible for users to modify the controller itself.  In a namespace-install the managed namespace should therefore not be the controller's namespace.
+
+You can typically further restrict what a user can do to just being able to submit workflows from templates using [the workflow requriments feature](workflow-requirements.md).
```

Comment:
Workflow Requirements is actually misnamed. The feature is actually called workflow restrictions. Can you, as a part of this PR, rename `workflow-requirements.md` to `workflow-restrictions.md `?

Also, I would word this as:
```
An administrator can further restrict what a user can do to only definitions of a WorkflowTemplate using [the workflow restrictions feature](workflow-restrictions.md).
```